spring-pulsar-spring-boot-autoconfigure/src/test/java/org/springframework/pulsar/autoconfigure/PulsarReactiveAutoConfigurationTests.java

```diff
@@ -0,0 +1,238 @@
+import org.junit.jupiter.api.Test;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+
+import org.springframework.boot.autoconfigure.AutoConfigurations;
+import org.springframework.boot.test.context.FilteredClassLoader;
+import org.springframework.boot.test.context.assertj.AssertableApplicationContext;
+import org.springframework.boot.test.context.runner.ApplicationContextRunner;
+import org.springframework.pulsar.config.PulsarClientFactoryBean;
+import org.springframework.pulsar.core.reactive.DefaultReactivePulsarSenderFactory;
+import org.springframework.pulsar.core.reactive.ReactivePulsarSenderFactory;
+import org.springframework.pulsar.core.reactive.ReactivePulsarSenderTemplate;
+
+/**
+ * Autoconfiguration tests for {@link PulsarReactiveAutoConfiguration}.
+ *
+ * @author Christophe Bornet
+ */
+@SuppressWarnings("unchecked")
+class PulsarReactiveAutoConfigurationTests {
+
+	private final ApplicationContextRunner contextRunner = new ApplicationContextRunner()
+			.withConfiguration(AutoConfigurations.of(PulsarAutoConfiguration.class))
+			.withConfiguration(AutoConfigurations.of(PulsarReactiveAutoConfiguration.class));
+
+	@Test
+	void autoConfigurationSkippedWhenReactivePulsarSenderTemplateNotOnClasspath() {
+		this.contextRunner.withClassLoader(new FilteredClassLoader(ReactivePulsarSenderTemplate.class)).run(
+				(context) -> assertThat(context).hasNotFailed().doesNotHaveBean(PulsarReactiveAutoConfiguration.class));
+	}
+
+	@Test
+	void defaultBeansAreAutoConfigured() {
+		this.contextRunner.run((context) -> assertThat(context).hasNotFailed()
+				.hasSingleBean(ReactivePulsarSenderTemplate.class).hasSingleBean(ReactivePulsarClient.class)
+				.hasSingleBean(ProducerCacheProvider.class).hasSingleBean(ReactiveMessageSenderCache.class)
+				.hasSingleBean(ReactivePulsarSenderFactory.class).getBean(ReactivePulsarSenderTemplate.class));
+	}
+
+	@Test
+	void customReactivePulsarSenderTemplateIsRespected() {
+		ReactivePulsarSenderTemplate<String> template = new ReactivePulsarSenderTemplate<>(null);
+		this.contextRunner
+				.withBean("customReactivePulsarSenderTemplate", ReactivePulsarSenderTemplate.class, () -> template)
+				.run((context) -> assertThat(context).hasNotFailed().getBean(ReactivePulsarSenderTemplate.class)
+						.isSameAs(template));
+	}
+
+	@Test
+	void customReactivePulsarClientIsRespected() {
+		ReactivePulsarClient client = AdaptedReactivePulsarClientFactory.create((PulsarClient) null);
+		this.contextRunner.withBean("customReactivePulsarClient", ReactivePulsarClient.class, () -> client).run(
+				(context) -> assertThat(context).hasNotFailed().getBean(ReactivePulsarClient.class).isSameAs(client));
+	}
+
+	@Test
+	void customProducerCacheProviderIsRespected() throws Exception {
+		try (ProducerCacheProvider provider = new ConcurrentHashMapProducerCacheProvider()) {
+			this.contextRunner
+					.withBean("customReactivePulsarSenderTemplate", ProducerCacheProvider.class, () -> provider)
```

Comment:
This (and a few tests below) name the bean "customReactivePulsarSenderTemplate" but its not that. :)

Reply:
done